pkg/sqlite/scene_filter.go

```diff
@@ -0,0 +1,337 @@
+
+	table := qb.table()
+	q := qb.selectDataset().Prepared(true).Where(table.Col(idColumn).In(ids))
+	unsorted, err := qb.getMany(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, s := range unsorted {
+		i := intslice.IntIndex(ids, int(s.ID))
+		ret[i] = s
+	}
+
+	for i := range ret {
+		if ret[i] == nil {
+			return nil, fmt.Errorf("scene filter with id %d not found", ids[i])
+		}
+	}
+
+	return ret, nil
+}
+
+func (qb *SceneFilterStore) find(ctx context.Context, id int) (*models.SceneFilter, error) {
+	q := qb.selectDataset().Where(qb.tableMgr.byID(id))
+
+	ret, err := qb.get(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+
+	return ret, nil
+}
+
+func (qb *SceneFilterStore) get(ctx context.Context, q *goqu.SelectDataset) (*models.SceneFilter, error) {
+	ret, err := qb.getMany(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(ret) == 0 {
+		return nil, sql.ErrNoRows
+	}
+
+	return ret[0], nil
+}
+
+func (qb *SceneFilterStore) getMany(ctx context.Context, q *goqu.SelectDataset) ([]*models.SceneFilter, error) {
+	const single = false
+	var ret []*models.SceneFilter
+	if err := queryFunc(ctx, q, single, func(r *sqlx.Rows) error {
+		var f sceneFilterRow
+		if err := r.StructScan(&f); err != nil {
+			return err
+		}
+
+		s := f.resolve()
+
+		ret = append(ret, s)
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return ret, nil
+}
+
+func (qb *SceneFilterStore) FindBySceneID(ctx context.Context, sceneID int) ([]*models.SceneFilter, error) {
+	query := `
+		SELECT scene_filters.* FROM scene_filters
+		WHERE scene_filters.scene_id = ?
+		GROUP BY scene_filters.id
+		ORDER BY scene_filters.id ASC
+	`
+	args := []interface{}{sceneID}
+	return qb.querySceneFilters(ctx, query, args)
+}
+
+func (qb *SceneFilterStore) makeFilter(ctx context.Context, sceneFilterFilter *models.SceneFilterFilterType) *filterBuilder {
+	query := &filterBuilder{}
+
+	query.handleCriterion(ctx, timestampCriterionHandler(sceneFilterFilter.CreatedAt, "scene_filters.created_at"))
+	query.handleCriterion(ctx, timestampCriterionHandler(sceneFilterFilter.UpdatedAt, "scene_filters.updated_at"))
+	query.handleCriterion(ctx, dateCriterionHandler(sceneFilterFilter.SceneDate, "scenes.date"))
+	query.handleCriterion(ctx, timestampCriterionHandler(sceneFilterFilter.SceneCreatedAt, "scenes.created_at"))
+	query.handleCriterion(ctx, timestampCriterionHandler(sceneFilterFilter.SceneUpdatedAt, "scenes.updated_at"))
+
+	return query
+}
+
+func (qb *SceneFilterStore) makeQuery(ctx context.Context, sceneFilterFilter *models.SceneFilterFilterType, findFilter *models.FindFilterType) (*queryBuilder, error) {
+	if sceneFilterFilter == nil {
+		sceneFilterFilter = &models.SceneFilterFilterType{}
+	}
+	if findFilter == nil {
+		findFilter = &models.FindFilterType{}
+	}
+
+	query := qb.newQuery()
+	distinctIDs(&query, sceneFilterTable)
+
+	if q := findFilter.Q; q != nil && *q != "" {
+		searchColumns := []string{"scene_filters.id", "scenes.id"}
```

Comment:
This is searching on ids, which is a bit strange. If you're going to use the filter `Q` at all, it should probably be on the scene title and not ids.